parser/parser_test.go

```diff
@@ -334,7 +338,7 @@ func TestFromFile(t *testing.T) {
 				endDataCondition := eventState.DataConditions[0]
 				assert.Equal(t, "notifycustomerworkflow", endDataCondition.End.ContinueAs.WorkflowID)
 				assert.Equal(t, "1.0", endDataCondition.End.ContinueAs.Version)
-				assert.Equal(t, "${ del(.customerCount) }", endDataCondition.End.ContinueAs.Data)
+				assert.Equal(t, model.Object{IObject: model.String("${ del(.customerCount) }")}, endDataCondition.End.ContinueAs.Data)
```

Comment:
I think you can consider creating builders and assessors for this interface, so that we avoid this kind of structures. Something like: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/intstr#IntOrString